pkg/mimirpb/mimir.pb.go

```diff
@@ -5912,19 +5913,25 @@ func (m *TimeSeriesRW2) MarshalToSizedBuffer(dAtA []byte) (int, error) {
 		}
 	}
 	if len(m.LabelsRefs) > 0 {
-		dAtA22 := make([]byte, len(m.LabelsRefs)*10)
+		// Modified code.
```

Comment:
Is this going to cause headaches the next time someone regenerates this file? It looks like the solution in the Prometheus repo opts to add a pair of `Optimized...` methods in a non-generated file.

Reply:
Hmm, good idea. Let's test it. There are already a massive amount of patches in this file, would be nice to cut down on them: https://github.com/grafana/mimir/blob/main/pkg/mimirpb/mimir.pb.go.expdiff

Reply:
In a branch off this PR https://github.com/grafana/mimir/compare/alexweav/marshal-optimize...alexweav/test-new-field-patch-conflicts:
- Adding a new type/editing anything that's not WriteRequest worked fine
- Changing WriteRequest (adding a new field) caused the patch to fail to apply

In a branch off main, same thing.

Looks like we can figure out what part the patch was having trouble by applying it with `git am -3 < pkg/mimirpb/mimir.pb.go.expdiff`. It failed on something related to `rw2PagedSymbols` in the `timeseries.Unmarshal()` call site, not the code added here - so I think this PR at least doesn't make it any worse than it is now.

Reply:
I realized a concern though - if we copy the marshal function, then add a new field, I'm not sure that the CI would tell you to go update the manual copy. The manual copy of the function would just compile fine, never do anything with that field, and just silently not send it? Maybe it failing loudly is a good thing?

Reply:
I forgot we already had a precedent for modifying these files.